### ISSUE TITLE
Switch to `cabal repl --build-depends=tidal` by default?

### ISSUE THREAD

**Issue by yaxu** [tidalcycles/atom-tidalcycles#156]
As per advice here: https://github.com/haskell/cabal-userguide/issues/1#issuecomment-862406613

This could work around the problems with the `cabal install tidal --lib` command. 

To run a repl we could do: `cabal repl --build-depends=tidal`

It is a bit slower than running `ghci` and prints out some confusing things about fake packages but works.

If tidal isn't installed then it will install it first, but only if `cabal update` has already been run. To build everything in advance we can just do `cabal install tidal` but then there is a heavy warning that doesn't apply to us.

**Comment by ndr-brt** [comment]
Interesting.
I tried to execute:
```
➜  Tidal git:(main) cabal update
Downloading the latest package list from hackage.haskell.org
Updated package list of hackage.haskell.org to the index-state 2021-06-17T03:31:32Z
To revert to previous state run:
    cabal v2-update 'hackage.haskell.org,2021-06-11T10:10:37Z'

➜  Tidal git:(main) cabal repl --build-depends=tidal              
Resolving dependencies...
Error:
    Dependency cycle between the following components: library
    In the inplace package 'tidal-1.7.7'
```

**Comment by ndr-brt** [comment]
ok, now it's working. Nice and clean, and all the output would be printed on the console inside atom.
Just a question, can we assert that everyone has `cabal` installed and in $PATH?
Or cabal path should be configurable as `ghci`?

**Comment by yaxu** [comment]
Yes I think cabal will be in the path, or at least in same folder as ghci.

**Comment by ndr-brt** [comment]
Do you know what's the minimum cabal version needed for this command?

**Comment by yaxu** [comment]
It looks like it's been there many years, but originally as `new-repl`. E.g. in 2.0: https://cabal.readthedocs.io/en/2.0/nix-local-build.html?highlight=new-repl#cabal-new-repl

The build-depends option isn't mentioned in the new-repl section until 2.4, but it might have worked before: https://cabal.readthedocs.io/en/2.4/nix-local-build.html#cabal-new-repl
